### PR TITLE
[EFR32] Adds fixes for IPv4/6 address resolution on RS911x and WF200

### PR DESCRIPTION
**Pull request by rosahay-silabs**
#### Problem
* Fix for address resolution timeout on RS9116 and WF200
* Fix for pull buffer empty issue.
* Refactors code with respect to the clang-format file (inside `third_party/EFR32`).

#### Change overview
Adds changes to fix the above stated problems.

#### Testing
How was this tested? (at least one bullet point required)
* If new unit tests are not added, why not?
  Since it is part of the commissioning flow and gets tested with that.

### CODE REVIEW

File: third_party/efr32_sdk/.clang-format [hunk 1]
```diff
@@ -0,0 +1,72 @@
+---
```
Comment (jmartinez-silabs) [inline third_party/efr32_sdk/.clang-format:1]: Why is this needed here?

**Review by jmartinez-silabs — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

File: src/platform/EFR32/CHIPDevicePlatformConfig.h [hunk 2]
```diff
@@ -48,17 +48,15 @@
 
 #define CHIP_DEVICE_CONFIG_ENABLE_CHIP_TIME_SERVICE_TIME_SYNC 0
 
-#if defined(RS911X_WIFI)
+#if defined(RS911X_WIFI) || defined(WF200_WIFI)
```
Comment (jmeg-sfy) [inline src/platform/EFR32/CHIPDevicePlatformConfig.h:51]: i m sure this pattern repeats often 
can't we think of a #define SI_MODULE_WIFI to gathering both modules ?

File: src/platform/EFR32/Logging.cpp [hunk 3]
```diff
@@ -133,10 +133,13 @@ extern "C" void efr32Log(const char * aFormat, ...)
...
     va_start(v, aFormat);
 #if EFR32_LOG_ENABLED
+    TickType_t now;
     char formattedMsg[CHIP_CONFIG_LOG_MESSAGE_MAX_SIZE];
     static_assert(sizeof(formattedMsg) > kMaxCategoryStrLen); // Greater than to at least accommodate a ending Null Character
 
-    strcpy(formattedMsg, LOG_EFR32);
+    // strcat(formattedMsg, LOG_EFR32);
+    now = xTaskGetTickCount();
+    snprintf(formattedMsg, CHIP_CONFIG_LOG_MESSAGE_MAX_SIZE, "[%ld] %s", now, LOG_EFR32);
```
Comment (jmeg-sfy) [inline src/platform/EFR32/Logging.cpp:142]: why do we need the timestamp
this gonna be crazy verbosy for the logs 

I would prefer a config flag for this and leave it by default as it was

**Review by jmeg-sfy — CHANGES_REQUESTED** [review]
check my comments 
also please add  EFR32 prefix to  your PR name, this helps identify the topics

thanks

File: src/platform/EFR32/Logging.cpp (hunk 3 above)
Comment (jmartinez-silabs) [reply src/platform/EFR32/Logging.cpp:142]: I am not totally opposed to it. Chip-tool and python controller have timestamps like that and it is still readable. Yes, you can add a flag to toggle it on and off yet. But I do see the appeal for debugging some timing-related issues.

File: third_party/efr32_sdk/.clang-format (hunk 1 above)
Comment (rosahay-silabs) [reply third_party/efr32_sdk/.clang-format:1]: Added to maintain consistency while porting files from other internal sources into MATTER.

File: src/platform/EFR32/CHIPDevicePlatformConfig.h (hunk 2 above)
Comment (rosahay-silabs) [reply src/platform/EFR32/CHIPDevicePlatformConfig.h]: using `#if defined(SL_WIFI)`

File: src/platform/EFR32/CHIPDevicePlatformConfig.h [hunk 4]
```diff
@@ -48,17 +48,15 @@
...
-#define CHIP_DEVICE_CONFIG_ENABLE_IPV6 1
-#else
+#else /* !WIFI_IPV4_DISABLED */
 #define CHIP_DEVICE_CONFIG_ENABLE_IPV4 1
-#define CHIP_DEVICE_CONFIG_ENABLE_IPV6 1
-#endif
+#endif /* WIFI_IPV4_DISABLED */
 
-#endif
+#endif /* SL_WIFI */
```
Comment (jmeg-sfy) [inline src/platform/EFR32/CHIPDevicePlatformConfig.h:60]: cool but where SL_WIFI is defined now ? directly into your SDK itself ? (third_party/efr32_sdk ?)

File: src/platform/EFR32/Logging.cpp (hunk 3 above)
Comment (rosahay-silabs) [reply src/platform/EFR32/Logging.cpp]: Reverted the change for now, will add a flag to toggle once we have a better timestamp resolution on logs.

Comment (jmeg-sfy) [reply src/platform/EFR32/Logging.cpp]: yes chip tool is handy but RTT debug is already crowded with logs for an embedded platform

File: src/platform/EFR32/CHIPDevicePlatformConfig.h (hunk 4 above)
Comment (rosahay-silabs) [reply src/platform/EFR32/CHIPDevicePlatformConfig.h:60]: Yeah its set in the GNI files, `SL_WIFI=1` for both WiFi modules

File: src/platform/EFR32/Logging.cpp (hunk 3 above)
Comment (rosahay-silabs) [reply src/platform/EFR32/Logging.cpp]: Yeah, I will make it a patch till we have a consensus. I feel remote scenarios might find this useful. My recommendation is to replace the `<tag >` with `<T>` and include timestamps instead.

**Review by jmeg-sfy — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]
